enforcer.go

```diff
@@ -281,10 +281,37 @@ func (e *Enforcer) ClearPolicy() {
 
 // LoadPolicy reloads the policy from file/database.
 func (e *Enforcer) LoadPolicy() error {
-	needToRebuild := false
-	newModel := e.model.Copy()
+	newModel, err := e.prepareNewModel(e.model.Copy())
+	if err != nil {
+		return err
+	}
+
+	return e.setNewModel(newModel)
+}
+
+// prepareNewModel prepares a new model with the policy from file/database.
+func (e *Enforcer) prepareNewModel(newModel model.Model) (model.Model, error) {
 	newModel.ClearPolicy()
 
+	if err := e.adapter.LoadPolicy(newModel); err != nil && err.Error() != "invalid file path, file path cannot be empty" {
+		return nil, err
+	}
+
+	if err := newModel.SortPoliciesBySubjectHierarchy(); err != nil {
+		return nil, err
+	}
+
+	if err := newModel.SortPoliciesByPriority(); err != nil {
+		return nil, err
+	}
+
+	return newModel, nil
+}
+
+// setNewModel replaces the current model with a new one
+func (e *Enforcer) setNewModel(newModel model.Model) error {
```

Comment:
This function is mainly used for build, and the name of this function will be confusing. Maybe you can change it to tryBuildRoleLinks or something. At the same time, extract 337 rows ``e.model=newModel``

Reply:
Updated, there seems to be actually no need for this function and a call to `BuildRoleLinks` is enough.